workflows/funcscan.nf

```diff
@@ -79,43 +67,117 @@ workflow FUNCSCAN {
-        .branch {
-            compressed: it[1].toString().endsWith('.gz')
-            uncompressed: it[1]
-        }
+    ch_input_prep = ch_input
+                        .map{meta, fasta, faa, gbk -> [meta, [fasta, faa, gbk]]}
+                        .transpose()
+                        .branch {
+                            compressed: it[1].toString().endsWith('.gz')
+                            uncompressed: it[1]
+                        }
 
-    GUNZIP_FASTA_PREP ( fasta_prep.compressed )
-    ch_versions = ch_versions.mix( GUNZIP_FASTA_PREP.out.versions )
+    GUNZIP_INPUT_PREP ( ch_input_prep.compressed )
+    ch_versions = ch_versions.mix( GUNZIP_INPUT_PREP.out.versions )
 
     // Merge all the already uncompressed and newly compressed FASTAs here into
     // a single input channel for downstream
-    ch_unzipped_fastas = GUNZIP_FASTA_PREP.out.gunzip
-                        .mix( fasta_prep.uncompressed )
+    ch_intermediate_input = GUNZIP_INPUT_PREP.out.gunzip
+                            .mix( ch_input_prep.uncompressed )
+                            .groupTuple()
+                            .map{
+                                meta, files ->
+                                    def fasta_found   = files.find{it.toString().tokenize('.').last().matches('fasta|fas|fna|fa')}
+                                    def faa_found     = files.find{it.toString().endsWith('.faa')}
+                                    def gbk_found     = files.find{it.toString().tokenize('.').last().matches('gbk')}
+                                    def fasta         = fasta_found   != null ? fasta_found   : []
+                                    def faa           = faa_found     != null ? faa_found     : []
+                                    def gbk           = gbk_found     != null ? gbk_found     : []
+
+                                    [meta, fasta, faa, gbk]
+                            }
+                            .branch {
+                                meta, fasta, faa, gbk ->
+                                    preannotated: gbk != []
+                                    fastas: true
+                            }
 
     // Split each FASTA into long and short contigs to
     // speed up BGC workflow with BGC-compatible contig lengths only
-    SEQKIT_SEQ_LONG ( ch_unzipped_fastas )
-    SEQKIT_SEQ_SHORT ( ch_unzipped_fastas )
+    ch_intermediate_fasta_for_split = ch_intermediate_input.fastas.map{ meta, fasta, faa, gbk -> [ meta, fasta ] }
+    SEQKIT_SEQ_LONG ( ch_intermediate_fasta_for_split )
+    SEQKIT_SEQ_SHORT ( ch_intermediate_fasta_for_split )
     ch_versions = ch_versions.mix( SEQKIT_SEQ_LONG.out.versions )
     ch_versions = ch_versions.mix( SEQKIT_SEQ_SHORT.out.versions )
 
-    ch_prepped_input_long = SEQKIT_SEQ_LONG.out.fastx
+    ch_intermediate_input_long = SEQKIT_SEQ_LONG.out.fastx
                                 .map{ meta, file -> [ meta + [id: meta.id + '_long', length: "long" ], file ] }
```

Comment:
Did you check the actual final output files (.tsv) with the `test_taxonomy.config` ? Do please correct me if im wrong with understanding the workflow now, so we are now renaming the meta.ids with suffixes `_long`and `_short`, will that not interfere with adding the taxonomy to the right files because taxonomy is added according to not only contig but sample_id too which comes from the meta_id.

Reply:
With regards to the summary and taxonomy :
- ampcombi: it doesnt output all samples (but that will be fixed when i put in the new ampcombi submodules). Taxonomy is merged
- combgc: it doesnt consider the samples_1 and 2 that are preannotated, only takes those from the annotatation step in the final report, which i dont understand why. Taxonomy is merged 
- hamronization: Works perfectly. 
Either therees a problem with this collectfile() function fro both ampcombi/combgc or somethings up with the publishdir path, that every time the pipeline stops and resumes it rewrites the final report.